internal/statemachine/tests_helper_test.go

```diff
@@ -0,0 +1,65 @@
+package statemachine
```

Comment:
I'm not 100% into go, but is this file name convention correct? I always thought `_test.go` files were for the actual tests, not test helpers.

Reply:
Files with `_test` suffix are excluded from regular package builds but will be included when the "go test" command is run. Otherwise you can put whatever functions you want in these files. Here I wanted to signal that what is in this file is used in the various tests across the package, and the tests only. I wanted to avoid the confusion with `helper_test.go` which is (by convention) a file that should contains functions to test what is in `helper.go` (and not contains test helpers).

If ubuntu-image had been destined to evolve more, I would have moved this to a dedicated "test" pkg at some point I think.